gobblin-dataset-config/src/main/java/gobblin/config/configstore/ConfigStore.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.configstore;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Collection;
+
+import com.typesafe.config.Config;
+
+/**
+ * A ConfigStore is the configuration store per scheme.
+ * 
+ * Each ConfigStore will have it's current version bundled 
+ * @author mitu
+ *
+ */
+public interface ConfigStore extends Serializable{
```

Comment:
The current version now is bundled with ConfigStore, that is why the getCurrentVersion() function appeared.